### ISSUE TITLE
Ensure chromium bots that were upgraded to macOS 13 have `safaridriver` enabled

### ISSUE THREAD

**Issue by vashworth** [flutter/flutter#137279]
### Is there an existing issue for this?

- [X] I have searched the [existing infra issues](https://github.com/flutter/flutter/issues?q=is%3Aopen+is%3Aissue+label%3Ateam-infra)

### Type of Request

bug

### Infrastructure Environment

LUCI

### What is happening?

Sometimes [web tests fail on macOS 13](https://github.com/flutter/flutter/issues/136972) and the web team suspect it may be because `safaridriver` needs to be enabled system wide.

I don't think there's a way to check if it's enabled via the command line. It can be check by opening Safari > Develop > Developer Settings and checking if "Allow remote automation" is checked. If the team doesn't have capability to check, than perhaps we can just enable it.

It can be enabled via the command line:
https://developer.apple.com/documentation/webkit/testing_with_webdriver_in_safari?language=objc

macOS 13 bots:
https://chromium-swarm.appspot.com/botlist?c=id&c=task&c=os&c=status&d=asc&f=os%3AMac-13&f=device_type%3Anone&k=device_type&s=id

**Comment by keyonghan** [comment]
We will need a bug to chromium team to enforce the change. Assigning to myself to link to the previous bug.

**Comment by keyonghan** [comment]
Created https://crbug.com/1496489 to track.

**Comment by keyonghan** [comment]
Added to ticket queue in case this issue could not be addressed by tomorrow. (I will be ooo next week)

**Comment by keyonghan** [comment]
@vashworth Could you give it a try with https://chromium-swarm.appspot.com/bot?id=mac-47-g572, which has been upgraded to mac-13.6.1? (https://crbug.com/1496489#c6)

**Comment by vashworth** [comment]
I confirmed https://github.com/flutter/flutter/issues/136972 is still happening: https://chromium-swarm.appspot.com/task?id=65d2b434c6db7910

**Comment by ricardoamador** [comment]
@eyebrowsoffire can you confirm you are looking into this issue?

**Comment by eyebrowsoffire** [comment]
I have not yet personally dug into the issue deeply. My cursory attempt at local repro was unsuccessful, it doesn't happen on my machine. However, the chrome infra folks had at least observed some of the flakiness when manually toying with safaridriver on the bots themselves. The chrome infra team in their bug (https://crbug.com/1496489) said they will be looking into the issue further this week. If they have more info about the failure, or some sort of workaround I could put into our test harness, I'm happy to make modifications. As-is, I can't repro locally and I don't personally have direct access to ssh into bots or anything like that, so I don't have enough info on my own to make progress.

**Comment by eyebrowsoffire** [comment]
Yes. I don't have any other way to move forward or investigate on my own.

**Comment by keyonghan** [comment]
Followed up in https://crbug.com/1496489 to see if any update.

**Comment by vashworth** [comment]
So I was able to recreate this issue on a devicelab bot (`flutter-devicelab-mac-11`). `safaridriver` was enabled.

@eyebrosoffire if you're unable to recreate locally, you should be able to recreate on `flutter-devicelab-mac-11`. You can run tests on it via led and you can VNC to it. I'll DM you some instructions on how to do this.

**Comment by vashworth** [comment]
> So I was able to recreate this issue on a devicelab bot (`flutter-devicelab-mac-11`). `safaridriver` was enabled.
> 
> @eyebrowsoffire if you're unable to recreate locally, you should be able to recreate on `flutter-devicelab-mac-11`. You can run tests on it via led and you can VNC to it. I'll DM you some instructions on how to do this.

So actually I take it back. I wasn't able to get this to work on a devicelab bot. It still times out but for a different reason. So we're back to square one.

For context: I enabled `safaridriver` manually on the bot with `sudo safaridriver --enable`,  but for some reason when I try to run the web tests on the bot it thinks `safaridriver` isn't enabled and gives error:

```
ERROR: safaridriver could not launch because it is not configured
[Webdriver][Error] correctly or you need to authenticate. Re-run safaridriver(1) and
[Webdriver][Error] pass the '--enable' flag to configure and/or authenticate.
[Webdriver][Error] For more information, consult the safaridriver(1) man page.
```
https://chromium-swarm.appspot.com/task?id=6659f1964adfad10

I also tried doing `safaridriver --enable` and typing password. I also tried specifying the user. I also tried making it's a recipe step (see in above task).

**Comment by keyonghan** [comment]
@vashworth Do we still need this considering we have the workaround?

**Comment by vashworth** [comment]
Technically `safaridriver` is still flakey and we're unable to recreate locally (so could be something specific to chromium bots). Also, the workaround does not appear to always work (https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20Production%20Engine%20Drone/231515/overview), but it does seem to work most of the time so I would consider this no longer critical or blocking, but still an issue.

**Comment by yjbanov** [comment]
Acknowledged.

**Comment by matanlurey** [comment]
Similar, does the iOS team want to own this (or close it)?

**Comment by vashworth** [comment]
We've been able to mitigate this be implementing retries, closing.
